app/views/posts/index.html.erb

```diff
@@ -1 +1,53 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<div class="center flex-column m-3">
+    <%= render partial: "shared/user", locals: {user: @user}%>
+    <div class="d-flex flex-column w-50 w-100-mobile">
+      <% if @user.posts.empty? %>
+        <p class="lead text-center">No posts yet!</p>
+      <% else %>
+        <% @user.posts.each do |post| %>
+          <div class="d-flex flex-column bg-light p-2 border mb-3">
+            <div class="d-flex align-items-center">
+              <h5 class="me-auto"><%= post.title %></h5>
+              <%= link_to "See post", "posts/#{post.id}", class: "btn btn-primary" %>
+            </div>
+            <p><%= post.text %></p>
+            <p class="small align-self-end">
+              Comments
+              <span class="badge bg-secondary">
+                <% if post.comments_counter %>
+                  <%= post.comments_counter %>
+                <% else %>
+                  0
+                <% end %>
+              </span>
+                Likes
+                <span class="badge bg-secondary">
+                  <% if post.likes_counter %>
+                    <%= post.likes_counter %>
+                  <% else %>
+                    0
+                  <% end %>
+            </p>
+            <div class="d-flex flex-column">
+              <h6>Recent comments</h6>
+              <ul class="group-item">
+                <% if post.comments.empty? %>
+                  <li class="list-group-item">No comments yet.</li>
+                <% else %>
+                  <% post.recent_comments.each do |comment| %>
+                    <li class="list-group-item">
+                      <span class="lead">
+                        <%= comment.author.name %>:
+                      </span>
+                    <%= comment.text %>
+                    </li>
+                  <% end %>
+                  <% end %>
+                </ul>
+              </div>
+            </div>
+          <% end %>
+        <% end %>
+      </div>
+        <%= button_to "Pagination", "", class: "btn btn-primary" %>
+    </div>
```

Comment:
- [ ] Please ensure that all the views are displayed without errors and ensure that the views are well associated with the Models and Controllers.

Reply:
- [ ] I am still not able to create a post in your post table you have `author_id` and `user_id` I am just confused about how you are creating a post, I think what you need is just a `user` who created a post, as I can see for your case foreign key should be `author_id` not ` user_id` please address that